testing-overhead/src/test/java/io/opentelemetry/containers/K6Container.java

```diff
@@ -14,22 +15,25 @@
 import org.testcontainers.containers.startupcheck.OneShotStartupCheckStrategy;
 import org.testcontainers.utility.DockerImageName;
 import org.testcontainers.utility.MountableFile;
+import java.nio.file.Path;
 import java.time.Duration;
 
 public class K6Container {
   private static final Logger logger = LoggerFactory.getLogger(K6Container.class);
   private final Network network;
   private final Agent agent;
   private final TestConfig config;
+  private final NamingConvention namingConvention;
 
-  public K6Container(Network network, Agent agent, TestConfig config) {
+  public K6Container(Network network, Agent agent, TestConfig config, NamingConvention namingConvention) {
     this.network = network;
     this.agent = agent;
     this.config = config;
+    this.namingConvention = namingConvention;
   }
 
   public GenericContainer<?> build(){
-    String k6OutputFile = "/results/k6_out_" + agent + ".json";
+    Path k6OutputFile = namingConvention.k6Results(agent);
```

Comment:
Nit: it'd be pretty good to hide all paths behind `NamingConvention`, for example

```java
.withFileSystemBind(localNamingConvention.rootDir(), containerNamingConvention.rootDir())
```

Reply:
I'm not sure about _all_, since there are a lot of moving parts and I could see it exploding pretty quickly, but I'm going to make this change (and the corresponding one in the petclinic container too) and we improve upon this over time if it feels useful.